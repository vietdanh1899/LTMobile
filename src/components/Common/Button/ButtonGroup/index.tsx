import React from 'react';
import QuickView from '@components/Common/View/QuickView';
import Button, { ButtonProps } from '../DefaultButton';
import FlatList, { FlatListProps } from '../../FlatList/DefaultFlatList';

export interface ButtonGroupProps extends Omit<ButtonProps, 'title'|'t'|'onPress'|'margin'|'marginLeft'|'marginVertical'>{
  propsChange?: boolean;
  titleList?: Array<string>;
  tList?: Array<string>;
  onItemPress?: (index: number) => any;
  flatListProps?: FlatListProps
  defaultActiveIndex?: number;
}

interface State {
  activeIndex: number;
}

class ButtonGroup extends React.Component<ButtonGroupProps, State> {
  static defaultProps = {
    propsChange: true,
  };

  private flatListRef: any;

  constructor(props: ButtonGroupProps) {
    super(props);
    const { defaultActiveIndex } = this.props;
    this.state = { activeIndex: defaultActiveIndex || 0 };
  }

  static getDerivedStateFromProps(nextProps: any, prevState: any) {
    const { propsChange } = nextProps;
    const { activeIndex } = prevState;
    if (propsChange && nextProps.defaultActiveIndex !== activeIndex) {
      return {
        activeIndex: nextProps.defaultActiveIndex,
      };
    }
    return {
      activeIndex,
    };
  }

  componentDidUpdate() {
    const { activeIndex } = this.state;
    if (this.flatListRef && activeIndex) {
      this.flatListRef.flatListRef.scrollToIndex({
        animated: true,
        index: activeIndex - 1 > 0 ? activeIndex - 1 : 0,
      });
    }
  }

  onPress = (index: number) => {
    this.setState({ activeIndex: index });
    const { onItemPress } = this.props;
    if (onItemPress) {
      onItemPress(index);
    }
  };

  getIndex = () => {
    const { activeIndex } = this.state;
    return activeIndex || 0;
  };

  renderItem = ({ item, index }: { item: string; index: number }) => {
    const {
      flatListProps,
      titleList,
      tList,
      marginRight: marginRightProp,
      marginHorizontal,
      ...otherButtonProps
    } = this.props;
    const { activeIndex } = this.state;
    const list = titleList || tList;
    const marginRight = marginRightProp || 10;
    if (list) {
      return (
        <Button
          outline
          {...otherButtonProps}
          title={titleList ? item : undefined}
          t={tList ? item : undefined}
          marginRight={
            index === list.length - 1 ? (marginHorizontal || 0) : marginRight
          }
          marginLeft={
            index === 0 ? marginHorizontal : 0
          }
          active={activeIndex === index}
          onPress={() => this.onPress(index)}
        />
      );
    }
    return <QuickView />;
  };

  render() {
    const {
      titleList, flatListProps, defaultActiveIndex, propsChange,
    } = this.props;
    const { activeIndex } = this.state;
    const FlatListButton: any = FlatList;
    return (
      <FlatListButton
        ref={(ref: any) => {
          this.flatListRef = ref;
        }}
        horizontal
        renderItem={this.renderItem}
        {...flatListProps}
        data={titleList}
        initialScrollIndex={defaultActiveIndex}
        onScrollToIndexFailed={() => {
          this.flatListRef.flatListRef.scrollToIndex({
            animated: true,
            index: 0,
          });
        }}
        showsHorizontalScrollIndicator={flatListProps?.horizontal || false}
        extraData={propsChange ? defaultActiveIndex : activeIndex}
      />
    );
  }
}

export default ButtonGroup;
