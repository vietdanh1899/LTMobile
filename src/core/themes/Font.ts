export const fontFamily = {
  RobotoBold: 'Roboto-Bold',
  RobotoItalic: 'Roboto-Italic',
  RobotoLight: 'Roboto-Light',
  RobotoRegular: 'Roboto-Regular',
  RobotoMedium: 'Roboto-Medium',
  RobotoThin: 'Roboto-Thin',
  GothamRoundedBold: 'GothamRounded-Bold',
  LatoBoldItalic: 'Lato-BoldItaclic',
  LatoBold: 'Lato-Bold',
};

export const fontWeight = {
  thin: '100',
  light: '300',
  medium: '600',
  bold: '700',
  heavy: '900',
};

export const fontSize = {
  tiny: 10,
  small: 12,
  medium: 14,
  large: 18,
  xLarge: 20,
};

export default { fontFamily, fontWeight, fontSize };
