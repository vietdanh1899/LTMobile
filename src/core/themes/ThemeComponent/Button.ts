import { Color, darkPrimaryColor, lightPrimaryColor } from './Common/Color';
import { roundedBorderRadius } from './Common/CommonProps';

/**
 * Button
 */
const borderWidth = 0;
const titleFontSize = 16;
const marginVertical = 5;
const marginHorizontal = 0;
const outlineBorderWidth = 1;

export const darkButton = {
  /**
   * Normal
   */
  borderWidth,
  borderColor: Color.white,
  backgroundColor: darkPrimaryColor,
  titleColor: Color.black,
  titleFontSize,
  marginVertical,
  marginHorizontal,
  roundedBorderRadius,
  /**
   * Outline
   */
  outlineBorderWidth,
  outlineBorderColor: darkPrimaryColor,
  outlineTitleColor: darkPrimaryColor,
  /**
   * Active
   */
  activeBackgroundColor: Color.blue,
  activeTitleColor: Color.white,
  activeOutlineTitleColor: Color.white,
  activeBorderColor: Color.blue,
  activeOutlineBorderColor: Color.blue,
};
export const lightButton = {
  /**
   * Normal
   */
  borderWidth,
  borderColor: Color.white,
  backgroundColor: lightPrimaryColor,
  titleColor: Color.white,
  titleFontSize,
  marginVertical,
  marginHorizontal,
  roundedBorderRadius,
  /**
   * Outline
   */
  outlineBorderWidth,
  outlineBorderColor: lightPrimaryColor,
  outlineTitleColor: lightPrimaryColor,
  /**
   * Active
   */
  activeBackgroundColor: Color.blue,
  activeTitleColor: Color.white,
  activeOutlineTitleColor: Color.white,
  activeBorderColor: Color.blue,
  activeOutlineBorderColor: Color.blue,
};
