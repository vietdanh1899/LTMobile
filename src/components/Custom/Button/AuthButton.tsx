import React, { PureComponent } from 'react';
import { withTheme, ThemeProps } from 'react-native-elements';
import Button, { ButtonProps } from '../../Common/Button/DefaultButton';

export interface AuthButtonProps extends ButtonProps{
}

class AuthButton extends PureComponent<AuthButtonProps> {
  static defaultProps = {
    type: 'solid',
    height: 50,
    width: '100%',
    bold: true,
    titlePaddingHorizontal: 20,
  };

  render() {
    const {
      theme,
      ...otherProps
    } = this.props;

    return (
      <Button
        {...otherProps}
      />
    );
  }
}

export default withTheme(
  AuthButton as unknown as React.ComponentType<AuthButtonProps & ThemeProps<any>>,
);
