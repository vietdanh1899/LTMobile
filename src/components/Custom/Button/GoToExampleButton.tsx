import React, { PureComponent } from 'react';
import NavigationService from '@utils/navigation';
import rootStack from '@contents/routes';
import Button from '../../Common/Button/DefaultButton';

class GoToExampleButton extends PureComponent {
  render() {
    return (
      <Button
        title="Go to Example"
        height={50}
        bold
        onPress={
        () => NavigationService.navigate(rootStack.exampleStack)
      }
      />
    );
  }
}

export default GoToExampleButton;
