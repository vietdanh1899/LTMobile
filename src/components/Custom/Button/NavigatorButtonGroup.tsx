import React from 'react';
import { TouchableOpacity } from 'react-native';
import Animated from 'react-native-reanimated';
import NavigationService from '@utils/navigation';
import ButtonGroup, { ButtonGroupProps } from '../../Common/Button/ButtonGroup';
import Text from '../../Common/Text';

export interface NavigatorButtonGroupProps extends ButtonGroupProps {
  state: any;
  position: any;
}

class NavigatorButtonGroup extends React.PureComponent<NavigatorButtonGroupProps> {
  onPress = (index: number) => {
    const { state } = this.props;
    const isFocused = state.index === index;
    if (!isFocused) {
      NavigationService.navigate(state.routes[index].name);
    }
  };

  renderItem = ({ item, index }: { item: string; index: number }) => {
    const { state, position } = this.props;
    const inputRange = state.routes.map((_: any, i: number) => i);
    const outputRange = inputRange.map((i: number) => (i === index ? 1 : 0));

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const opacity = Animated.interpolate(position, {
      inputRange,
      outputRange,
    });
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const color = Animated.color(
      Animated.round(
        Animated.interpolate(position, {
          inputRange,
          outputRange: inputRange.map((inputIndex: any) => (inputIndex === index ? 255 : 0)),
        }),
      ),
      0,
      0,
    );
    const backgroundColor = Animated.color(
      Animated.round(
        Animated.interpolate(position, {
          inputRange,
          outputRange: inputRange.map((inputIndex: any) => (inputIndex === index ? 1 : 230)),
        }),
      ),
      Animated.round(
        Animated.interpolate(position, {
          inputRange,
          outputRange: inputRange.map((inputIndex: any) => (inputIndex === index ? 32 : 233)),
        }),
      ),
      Animated.round(
        Animated.interpolate(position, {
          inputRange,
          outputRange: inputRange.map((inputIndex: any) => (inputIndex === index ? 102 : 240)),
        }),
      ),
    );
    return (
      <Animated.View style={{ backgroundColor }}>
        <TouchableOpacity
          accessibilityRole="button"
          onPress={() => {
            const isFocused = state.index === index;
            if (!isFocused) {
              NavigationService.navigate(state.routes[index].name);
            }
          }}
        >
          <Text color="#FFFFFF">{item}</Text>
        </TouchableOpacity>
      </Animated.View>
    );
  };

  render() {
    const { state, position, ...otherProps } = this.props;
    return (
      <ButtonGroup
        {...otherProps}
        defaultActiveIndex={state.index}
        flatListProps={{ renderItem: this.renderItem }}
      />
    );
  }
}

export default NavigatorButtonGroup;
