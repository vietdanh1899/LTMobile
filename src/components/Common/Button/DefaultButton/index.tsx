import React from 'react';
import { StyleSheet, Platform } from 'react-native';
import {
  Button as ElementButton,
  ButtonProps as EButtonProps,
  withTheme,
  ThemeProps,
} from 'react-native-elements';
import Font from '@themes/Font';
import _ from 'lodash';
import { Translation } from 'react-i18next';
import { Color } from '@themes/Theme';

export interface ButtonProps extends Omit<EButtonProps, 'raised' | 'type'> {
  width?: number | string;
  height?: number | string;
  margin?: number;
  marginTop?: number;
  marginBottom?: number;
  marginLeft?: number;
  marginRight?: number;
  marginHorizontal?: number;
  marginVertical?: number;
  padding?: number;
  titlePadding?: number;
  titlePaddingVertical?: number;
  titlePaddingHorizontal?: number;
  paddingTop?: number;
  paddingBottom?: number;
  paddingLeft?: number;
  paddingRight?: number;
  paddingHorizontal?: number;
  paddingVertical?: number;
  primary?: boolean;
  secondary?: boolean;
  success?: boolean;
  warning?: boolean;
  error?: boolean;
  title?: string;
  titleCenter?: boolean;
  titleColor?: string;
  backgroundColor?: string;
  borderRadius?: number;
  borderColor?: string;
  activeBorderColor?: string;
  borderWidth?: number;
  center?: boolean;
  bold?: boolean;
  fontSize?: number;
  active?: boolean;
  activeBackgroundColor?: string;
  activeTitleColor?: string;
  outline?: boolean;
  clear?: boolean;
  sharp?: boolean;
  rounded?: boolean;
  circle?: boolean;
  shadow?: boolean;
  color?: string;
  reverseColor?: boolean;
  theme?: any;
  t?: string;
}

class Button extends React.PureComponent<ButtonProps> {
  static defaultProps = {
    padding: 0,
    margin: 0,
    active: false,
    rounded: true,
    titleCenter: true,
  };

  render() {
    const {
      width: widthProp,
      height: heightProp,
      margin,
      marginTop,
      marginBottom,
      marginLeft,
      marginRight,
      marginHorizontal: marginHorizontalProp,
      marginVertical: marginVerticalProp,
      padding,
      titlePadding,
      titlePaddingVertical,
      titlePaddingHorizontal,
      paddingTop,
      paddingBottom,
      paddingLeft,
      paddingRight,
      paddingHorizontal,
      paddingVertical,
      borderRadius: borderRadiusProp,
      primary,
      secondary,
      success,
      error,
      warning,
      borderColor: borderColorProp,
      activeBorderColor: activeBorderColorProp,
      borderWidth,
      center,
      bold,
      fontSize,
      active,
      icon: iconProp,
      iconRight,
      backgroundColor: backgroundColorProp,
      title,
      titleCenter,
      titleColor: titleColorProp,
      activeBackgroundColor: activeBackgroundColorProp,
      activeTitleColor: activeTitleColorProp,
      containerStyle: containerStyleProp,
      buttonStyle: buttonStyleProp,
      titleStyle: titleStyleProp,
      iconContainerStyle: iconContainerStyleProp,
      outline,
      clear,
      sharp: sharpProp,
      rounded: roundedProp,
      circle: circleProp,
      shadow,
      color,
      reverseColor,
      theme,
      t,
      ...otherProps
    } = this.props;

    let colors: any;
    if (Platform.OS === 'ios') {
      colors = theme.colors.platform.ios;
    } else {
      colors = theme.colors.platform.android;
    }

    /**
     * containerStyle
     */
    const marginVertical = marginVerticalProp || theme.Button.marginVertical;
    const marginHorizontal = marginHorizontalProp || theme.Button.marginHorizontal;
    let width = widthProp;
    let height = heightProp;

    let borderRadius: any = 0;
    let sharp = sharpProp;
    let rounded = roundedProp;
    let circle = circleProp;
    if (borderRadiusProp) {
      borderRadius = borderRadiusProp;
    } else {
      if (circle) {
        rounded = false;
        sharp = false;
        const minDimension = _.min([width, height]) || 50;
        width = minDimension;
        height = minDimension;
        borderRadius = minDimension;
      }
      if (rounded) {
        borderRadius = theme.Button.roundedBorderRadius;
      }
      if (sharp) {
        rounded = false;
        circle = false;
        borderRadius = 0;
      }
    }
    const { shadowView } = theme;

    const containerStyle: any = StyleSheet.flatten([
      {
        margin,
        marginTop,
        marginBottom,
        marginLeft,
        marginRight,
        marginHorizontal,
        marginVertical,
        padding,
        paddingTop,
        paddingLeft,
        paddingRight,
        paddingHorizontal,
        paddingVertical,
        borderRadius,
      },
      shadow && { paddingBottom: paddingBottom || 3 },
      center && { alignSelf: 'center' },
      containerStyleProp,
    ]);

    /**
     * buttonStyle
     */
    const activeBackgroundColor = activeBackgroundColorProp
    || color
    || theme.Button.activeBackgroundColor;
    let backgroundColor = backgroundColorProp || color || theme.Button.backgroundColor;

    if (primary) backgroundColor = colors.primary;
    if (secondary) backgroundColor = colors.secondary;
    if (success) backgroundColor = colors.success;
    if (warning) backgroundColor = colors.warning;
    if (error) backgroundColor = colors.error;

    const activeBorderColor = activeBorderColorProp || color || theme.Button.activeBorderColor;
    let borderColor = borderColorProp || color || theme.Button.borderColor;
    if (primary) borderColor = colors.primary;
    if (secondary) borderColor = colors.secondary;
    if (success) borderColor = colors.success;
    if (warning) borderColor = colors.warning;
    if (error) borderColor = colors.error;

    let buttonStyle = StyleSheet.flatten([
      {
        width,
        height,
        borderRadius,
        borderWidth: borderWidth || theme.Button.borderWidth,
        borderColor: active ? activeBorderColor : borderColor,
        backgroundColor: active ? activeBackgroundColor : backgroundColor,
      },
      titlePadding && { padding: titlePadding },
      titlePaddingHorizontal && { paddingHorizontal: titlePaddingHorizontal },
      titlePaddingVertical && { paddingVertical: titlePaddingVertical },
      shadow && shadowView,
      buttonStyleProp,
      !titleCenter && { justifyContent: 'flex-start' },
    ]);

    /**
     * titleStyle
     */

    const activeTitleColor = activeTitleColorProp
    || (color ? Color.white : theme.Button.activeTitle);
    const titleColor = titleColorProp || (color ? Color.white : theme.Button.titleColor);

    let titleStyle: any = StyleSheet.flatten([
      {
        fontWeight: (active || bold) ? Font.fontWeight.bold : Font.fontWeight.medium,
        fontSize: fontSize || theme.Button.titleFontSize,
        color: active ? activeTitleColor : titleColor,
        marginRight: (iconProp && !iconRight && !circle) ? 10 : 0,
        marginLeft: (iconProp && iconRight && !circle) ? 10 : -10,
        paddingHorizontal: 5,
      },
      titleStyleProp,
    ]);

    /**
     * icon, iconContainerStyle
     */
    const iconContainerStyle: any = StyleSheet.flatten([
      (iconProp && !iconRight && title) && { marginRight: 15 },
      iconContainerStyleProp,
    ]);
    const defaultIcon = {
      size: 20,
      color: active ? activeTitleColor : titleColor,
      type: 'material-community',
    };
    const icon = _.merge(defaultIcon, iconProp);

    /**
     * type
     */
    if (clear) {
      buttonStyle = StyleSheet.flatten([
        buttonStyle,
        {
          backgroundColor: 'transparent',
          borderWidth: 0,
        },
      ]);

      const activeOutlineTitleColor = activeTitleColorProp
      || color
      || theme.Button.activeOutlineTitleColor;
      let outlineTitleColor = titleColorProp || color || theme.Button.outlineTitleColor;
      if (primary) outlineTitleColor = colors.primary;
      if (secondary) outlineTitleColor = colors.secondary;
      if (success) outlineTitleColor = colors.success;
      if (warning) outlineTitleColor = colors.warning;
      if (error) outlineTitleColor = colors.error;

      titleStyle = StyleSheet.flatten([
        titleStyle,
        {
          color: active ? activeOutlineTitleColor : outlineTitleColor,
        },
      ]);
    } else if (outline) {
      const activeOutlineBorderColor = activeBorderColorProp
      || color
      || theme.Button.activeOutlineBorderColor;
      let outlineBorderColor = borderColorProp || color || theme.Button.outlineBorderColor;
      if (primary) outlineBorderColor = colors.primary;
      if (secondary) outlineBorderColor = colors.secondary;
      if (success) outlineBorderColor = colors.success;
      if (warning) outlineBorderColor = colors.warning;
      if (error) outlineBorderColor = colors.error;

      const activeOutlineTitleColor = activeTitleColorProp
      || color
      || theme.Button.activeOutlineTitleColor;
      let outlineTitleColor = titleColorProp || color || theme.Button.outlineTitleColor;
      if (primary) outlineTitleColor = colors.primary;
      if (secondary) outlineTitleColor = colors.secondary;
      if (success) outlineTitleColor = colors.success;
      if (warning) outlineTitleColor = colors.warning;
      if (error) outlineTitleColor = colors.error;

      buttonStyle = StyleSheet.flatten([
        buttonStyle,
        {
          backgroundColor: active ? activeBackgroundColor : 'transparent',
          borderWidth: borderWidth || theme.Button.outlineBorderWidth,
          borderColor: active ? activeOutlineBorderColor : outlineBorderColor,
        },
      ]);

      titleStyle = StyleSheet.flatten([
        titleStyle,
        {
          color: active ? activeOutlineTitleColor : outlineTitleColor,
        },
      ]);
    }

    /**
     * reverseColor
     */
    if (reverseColor) {
      const reverseBackgroundColor = titleStyle.color;
      const reverseTitleColor = buttonStyle.backgroundColor;
      buttonStyle.backgroundColor = reverseBackgroundColor;
      titleStyle.color = reverseTitleColor;
    }

    if (t) {
      return (
        <Translation>
          {
          (trans) => (
            <ElementButton
              {...otherProps}
              icon={icon}
              iconRight={iconRight}
              containerStyle={containerStyle}
              buttonStyle={buttonStyle}
              title={trans(t)}
              titleStyle={titleStyle}
              iconContainerStyle={iconContainerStyle}
            />
          )
          }
        </Translation>
      );
    }
    return (
      <ElementButton
        {...otherProps}
        icon={icon}
        iconRight={iconRight}
        containerStyle={containerStyle}
        buttonStyle={buttonStyle}
        title={title}
        titleStyle={titleStyle}
        iconContainerStyle={iconContainerStyle}
      />
    );
  }
}

export default withTheme(Button as React.ComponentType<ButtonProps & ThemeProps<any>>);
