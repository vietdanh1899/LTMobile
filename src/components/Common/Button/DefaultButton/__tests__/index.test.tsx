import React from 'react';
import Font from '@themes/Font';
import {
  basicTest, testStyleProps, getStyle, EnumComponent,
} from '@utils/testHelper';
import { lightTheme } from '@themes';
import { Platform } from 'react-native';
import { Color } from '@themes/Theme';
import Button from '..';

const target = <Button />;
const type = EnumComponent.HOC;

describe('Button Component', () => {
  basicTest(target, type);

  it('should apply all style-related props', () => {
    const props = {
      width: 100,
      height: 100,
      margin: 10,
      marginVertical: 10,
      marginHorizontal: 10,
      marginTop: 10,
      marginBottom: 10,
      marginLeft: 10,
      marginRight: 10,
      padding: 10,
      paddingVertical: 10,
      paddingHorizontal: 10,
      paddingTop: 10,
      paddingBottom: 10,
      paddingLeft: 10,
      paddingRight: 10,
    };
    testStyleProps(target, props, type, 'containerStyle');

    const props1 = {
      borderRadius: 20,
      borderColor: Color.orange,
      borderWidth: 5,
    };
    testStyleProps(target, props1, type, 'buttonStyle');
  });

  it('should apply all custom props', () => {
    /**
     * center
     */
    const testStyle = getStyle(target, { center: true }, type, 'containerStyle');
    expect(testStyle.alignSelf).toBe('center');

    /**
     * bold, titleColor
     */
    const testStyle1 = getStyle(target, { bold: true }, type, 'titleStyle');
    expect(testStyle1.fontWeight).toBe(Font.fontWeight.bold);
    const testStyle2 = getStyle(target, { titleColor: Color.orange }, type, 'titleStyle');
    expect(testStyle2.color).toBe(Color.orange);

    /**
     * type = solid | null, primary, secondary, success, warning, error, backgroundColor,
     */
    Platform.OS = 'android';
    const testStyle3 = getStyle(target, { primary: true }, type, 'buttonStyle');
    expect(testStyle3.backgroundColor).toBe(lightTheme.colors.platform.android.primary);
    const testStyle4 = getStyle(target, { secondary: true }, type, 'buttonStyle');
    expect(testStyle4.backgroundColor).toBe(lightTheme.colors.platform.android.secondary);
    const testStyle5 = getStyle(target, { success: true }, type, 'buttonStyle');
    expect(testStyle5.backgroundColor).toBe(lightTheme.colors.platform.android.success);
    const testStyle6 = getStyle(target, { error: true }, type, 'buttonStyle');
    expect(testStyle6.backgroundColor).toBe(lightTheme.colors.platform.android.error);
    const testStyle7 = getStyle(target, { warning: true }, type, 'buttonStyle');
    expect(testStyle7.backgroundColor).toBe(lightTheme.colors.platform.android.warning);
    const testStyle8 = getStyle(target, { backgroundColor: Color.orange }, type, 'buttonStyle');
    expect(testStyle8.backgroundColor).toBe(Color.orange);
    jest.resetModules();

    Platform.OS = 'ios';
    const testStyle9 = getStyle(target, { primary: true }, type, 'buttonStyle');
    expect(testStyle9.backgroundColor).toBe(lightTheme.colors.platform.ios.primary);
    const testStyle10 = getStyle(target, { secondary: true }, type, 'buttonStyle');
    expect(testStyle10.backgroundColor).toBe(lightTheme.colors.platform.ios.secondary);
    const testStyle11 = getStyle(target, { success: true }, type, 'buttonStyle');
    expect(testStyle11.backgroundColor).toBe(lightTheme.colors.platform.ios.success);
    const testStyle12 = getStyle(target, { error: true }, type, 'buttonStyle');
    expect(testStyle12.backgroundColor).toBe(lightTheme.colors.platform.ios.error);
    const testStyle13 = getStyle(target, { warning: true }, type, 'buttonStyle');
    expect(testStyle13.backgroundColor).toBe(lightTheme.colors.platform.ios.warning);
    const testStyle14 = getStyle(target, { backgroundColor: Color.orange }, type, 'buttonStyle');
    expect(testStyle14.backgroundColor).toBe(Color.orange);
    jest.resetModules();

    /**
     * type = outline, primary, secondary, success, warning, error,
     */
    const testStyle15 = getStyle(target, { outline: true }, type, 'buttonStyle');
    expect(testStyle15.backgroundColor).toBe('transparent');
    expect(testStyle15.borderColor).toBe(lightTheme.Button.outlineBorderColor);

    Platform.OS = 'android';
    const testStyleOutline3 = getStyle(target, { outline: true, primary: true }, type, 'buttonStyle');
    expect(testStyleOutline3.borderColor).toBe(lightTheme.colors.platform.android.primary);
    const testStyleOutline4 = getStyle(target, { outline: true, secondary: true }, type, 'buttonStyle');
    expect(testStyleOutline4.borderColor).toBe(lightTheme.colors.platform.android.secondary);
    const testStyleOutline5 = getStyle(target, { outline: true, success: true }, type, 'buttonStyle');
    expect(testStyleOutline5.borderColor).toBe(lightTheme.colors.platform.android.success);
    const testStyleOutline6 = getStyle(target, { outline: true, error: true }, type, 'buttonStyle');
    expect(testStyleOutline6.borderColor).toBe(lightTheme.colors.platform.android.error);
    const testStyleOutline7 = getStyle(target, { outline: true, warning: true }, type, 'buttonStyle');
    expect(testStyleOutline7.borderColor).toBe(lightTheme.colors.platform.android.warning);
    jest.resetModules();

    Platform.OS = 'ios';
    const testStyleOutline9 = getStyle(target, { outline: true, primary: true }, type, 'buttonStyle');
    expect(testStyleOutline9.borderColor).toBe(lightTheme.colors.platform.ios.primary);
    const testStyleOutline10 = getStyle(target, { outline: true, secondary: true }, type, 'buttonStyle');
    expect(testStyleOutline10.borderColor).toBe(lightTheme.colors.platform.ios.secondary);
    const testStyleOutline11 = getStyle(target, { outline: true, success: true }, type, 'buttonStyle');
    expect(testStyleOutline11.borderColor).toBe(lightTheme.colors.platform.ios.success);
    const testStyleOutline12 = getStyle(target, { outline: true, error: true }, type, 'buttonStyle');
    expect(testStyleOutline12.borderColor).toBe(lightTheme.colors.platform.ios.error);
    const testStyleOutline13 = getStyle(target, { outline: true, warning: true }, type, 'buttonStyle');
    expect(testStyleOutline13.borderColor).toBe(lightTheme.colors.platform.ios.warning);
    jest.resetModules();

    /**
     * type = clear, primary, secondary, success, warning, error,
     */
    const testStyle16 = getStyle(target, { clear: true }, type, 'buttonStyle');
    expect(testStyle16.backgroundColor).toBe('transparent');
    expect(testStyle16.borderWidth).toBe(0);
    const testStyle17 = getStyle(target, { clear: true }, type, 'titleStyle');
    expect(testStyle17.color).toBe(lightTheme.Button.outlineTitleColor);

    Platform.OS = 'android';
    const testStyleClear3 = getStyle(target, { clear: true, primary: true }, type, 'titleStyle');
    expect(testStyleClear3.color).toBe(lightTheme.colors.platform.android.primary);
    const testStyleClear4 = getStyle(target, { clear: true, secondary: true }, type, 'titleStyle');
    expect(testStyleClear4.color).toBe(lightTheme.colors.platform.android.secondary);
    const testStyleClear5 = getStyle(target, { clear: true, success: true }, type, 'titleStyle');
    expect(testStyleClear5.color).toBe(lightTheme.colors.platform.android.success);
    const testStyleClear6 = getStyle(target, { clear: true, error: true }, type, 'titleStyle');
    expect(testStyleClear6.color).toBe(lightTheme.colors.platform.android.error);
    const testStyleClear7 = getStyle(target, { clear: true, warning: true }, type, 'titleStyle');
    expect(testStyleClear7.color).toBe(lightTheme.colors.platform.android.warning);
    jest.resetModules();

    Platform.OS = 'ios';
    const testStyleClear9 = getStyle(target, { clear: true, primary: true }, type, 'titleStyle');
    expect(testStyleClear9.color).toBe(lightTheme.colors.platform.ios.primary);
    const testStyleClear10 = getStyle(target, { clear: true, secondary: true }, type, 'titleStyle');
    expect(testStyleClear10.color).toBe(lightTheme.colors.platform.ios.secondary);
    const testStyleClear11 = getStyle(target, { clear: true, success: true }, type, 'titleStyle');
    expect(testStyleClear11.color).toBe(lightTheme.colors.platform.ios.success);
    const testStyleClear12 = getStyle(target, { clear: true, error: true }, type, 'titleStyle');
    expect(testStyleClear12.color).toBe(lightTheme.colors.platform.ios.error);
    const testStyleClear13 = getStyle(target, { clear: true, warning: true }, type, 'titleStyle');
    expect(testStyleClear13.color).toBe(lightTheme.colors.platform.ios.warning);
    jest.resetModules();

    /**
     * active, activeBorderColor, activeBackgroundColor, activeTitleColor
     */
    const testStyle18 = getStyle(target, {
      active: 'true', activeBorderColor: Color.orange, activeBackgroundColor: Color.orange,
    }, type, 'buttonStyle');
    expect(testStyle18.borderColor).toBe(Color.orange);
    expect(testStyle18.backgroundColor).toBe(Color.orange);

    const testStyle19 = getStyle(target, {
      active: 'true', activeTitleColor: Color.orange,
    }, type, 'titleStyle');
    expect(testStyle19.color).toBe(Color.orange);

    /**
     * sharp, rounded, circle
     */
    const testStyle20 = getStyle(target, { sharp: 'true' }, type, 'buttonStyle');
    expect(testStyle20.borderRadius).toBe(0);

    const testStyle21 = getStyle(target, { rounded: 'true' }, type, 'buttonStyle');
    expect(testStyle21.borderRadius).toBe(lightTheme.Button.roundedBorderRadius);

    const testStyle22 = getStyle(target, { circle: 'true' }, type, 'buttonStyle');
    expect(testStyle22.width).toBe(testStyle22.height);
    expect(testStyle22.width).toBe(testStyle22.borderRadius);

    /**
     * shadow
     */
    const testStyle23 = getStyle(target, { shadow: true }, type, 'buttonStyle');
    expect(testStyle23).toMatchObject(lightTheme.shadowView);
    const testStyle24 = getStyle(target, { shadow: true }, type, 'containerStyle');
    expect(testStyle24.paddingBottom).toBeGreaterThanOrEqual(3);
  });
});
