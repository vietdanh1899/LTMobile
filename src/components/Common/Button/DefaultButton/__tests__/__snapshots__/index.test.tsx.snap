// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render without issues 1`] = `
<ForwardRef(Themed.Button)
  buttonStyle={
    Object {
      "backgroundColor": "#5760EB",
      "borderColor": "#ffffff",
      "borderRadius": 20,
      "borderWidth": 0,
      "height": undefined,
      "width": undefined,
    }
  }
  containerStyle={
    Object {
      "borderRadius": 20,
      "margin": 0,
      "marginBottom": undefined,
      "marginHorizontal": 0,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "marginVertical": 5,
      "padding": 0,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
    }
  }
  icon={
    Object {
      "color": "#ffffff",
      "size": 20,
      "type": "material-community",
    }
  }
  iconContainerStyle={Object {}}
  titleStyle={
    Object {
      "color": "#ffffff",
      "fontSize": 16,
      "fontWeight": "600",
      "marginLeft": -10,
      "marginRight": 0,
      "paddingHorizontal": 5,
    }
  }
/>
`;
